tracing implementation

bjoernschroeder opened the pull request.

bjoernschroeder committed:
implement tracing

bjoernschroeder committed:
fix inheritance, implement constructor, use string for context

bjoernschroeder committed:
add ros, finish MasterPlan trace

bjoernschroeder committed:
remove old code

bjoernschroeder committed:
change license to MIT

veerajsk commented on alica_tracing/include/tracing/AlicaTrace.h:
```diff
@@ -0,0 +1,197 @@
+            _rawTrace = opentracing::Tracer::Global()->StartSpan(opName, {opentracing::ChildOf(&parentCtx)});
+        } else {
+            _rawTrace = opentracing::Tracer::Global()->StartSpan(opName);
+            if (opName == "MasterPlan") {
```
put a comment here that this is required so that the traces are reported immediately and not when the master plan terminates which is basically only when the process dies

veerajsk commented on alica_tracing/include/tracing/AlicaTrace.h:
```diff
@@ -0,0 +1,197 @@
+
+using TracePtr = std::unique_ptr<Trace>;
+
+template <class TraceType>
```
Don't see any use in making this a template because this factory can't create traces of any other type.

veerajsk commented on alica_tracing/include/tracing/AlicaTrace.h:
```diff
@@ -0,0 +1,197 @@
+    // configFilePath is the place factory will look out for tracing configuration.
+    // returns false if initialization failed, true otherwise
+    // Will throw if the initialization failed
+    void init(const std::string& serviceName, const std::string& configFilePath)
```
need to define a constructor that takes these parameters & does the initilization. The constructor should also take in the default tags as a parameter & set them after the initialization. Otherwise it won't be possible for the AlicaContext to create the factory instance internally.

veerajsk commented on alica_tracing/include/tracing/AlicaTrace.h:
```diff
@@ -0,0 +1,197 @@
+
+private:
+    bool _initialized = false;
+    std::unordered_map<std::string, std::string> _defaultTags;
```
it's ok to store RawTraceValue here. The limitation of strings only applies to IAlicaTrace & we should extend it in the future anyway

veerajsk commented on alica_tracing/src/tracing/AlicaTrace.cpp:
```diff
@@ -0,0 +1 @@
+#include "tracing/AlicaTrace.h"
```
A template class should be defined in the header & so this file is not required. However since the TraceFactory need not be a template anymore, you can move all the definitions of AlicaTrace & TraceFactory to this file & only have the declarations in the header

veerajsk commented on alica_tracing/include/tracing/AlicaTrace.h:
```diff
@@ -0,0 +1,197 @@
+
+#include <engine/IAlicaTrace.h>
+#include <jaegertracing/Tracer.h>
+#include <ros/ros.h>
```
do we need this?

bjoernschroeder committed:
split header and cpp files, remove templates, remove setDefaultTags, pass defaultTags via constructor

bjoernschroeder committed:
use rawTraceValue

veerajsk committed:
Merge branch 'rr-devel' into bas_tracing

veerajsk committed:
Fix tracing

- Store the context in the trace because the master trace is deleted
immediately
- Delete the master trace after setting the default tags on it

veerajsk commented:
Replaced by https://github.com/rapyuta-robotics/alica-supplementary/pull/55
